python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py

```diff
@@ -596,43 +629,69 @@ def resolve_pipelineRunTags(self, graphene_info: ResolveInfo):
     def resolve_runGroupOrError(self, graphene_info: ResolveInfo, runId):
         return get_run_group(graphene_info, runId)
 
-    def resolve_isPipelineConfigValid(self, graphene_info: ResolveInfo, pipeline, **kwargs):
+    def resolve_isPipelineConfigValid(
+        self,
+        graphene_info: ResolveInfo,
+        pipeline: GraphenePipelineSelector,
+        mode: str,
+        runConfigData: Optional[GrapheneRunConfigData] = None,
+    ):
         return validate_pipeline_config(
             graphene_info,
             pipeline_selector_from_graphql(pipeline),
-            parse_run_config_input(kwargs.get("runConfigData", {}), raise_on_error=False),
-            kwargs.get("mode"),
+            parse_run_config_input(runConfigData or {}, raise_on_error=False),  # type: ignore
+            mode,
         )
 
-    def resolve_executionPlanOrError(self, graphene_info: ResolveInfo, pipeline, **kwargs):
+    def resolve_executionPlanOrError(
+        self,
+        graphene_info: ResolveInfo,
+        pipeline: GraphenePipelineSelector,
+        mode: str,
+        runConfigData: Optional[GrapheneRunConfigData] = None,
+    ):
         return get_execution_plan(
             graphene_info,
             pipeline_selector_from_graphql(pipeline),
-            parse_run_config_input(kwargs.get("runConfigData", {}), raise_on_error=True),  # type: ignore
-            kwargs.get("mode"),
+            parse_run_config_input(runConfigData or {}, raise_on_error=True),  # type: ignore
+            mode,
         )
 
-    def resolve_runConfigSchemaOrError(self, graphene_info: ResolveInfo, **kwargs):
+    def resolve_runConfigSchemaOrError(
+        self,
+        graphene_info: ResolveInfo,
+        selector: GraphenePipelineSelector,
+        mode: Optional[str] = None,
+    ):
         return resolve_run_config_schema_or_error(
             graphene_info,
-            pipeline_selector_from_graphql(kwargs["selector"]),
-            kwargs.get("mode"),
+            pipeline_selector_from_graphql(selector),
+            mode,
         )
 
     def resolve_instance(self, graphene_info: ResolveInfo):
         return GrapheneInstance(graphene_info.context.instance)
 
-    def resolve_assetNodes(self, graphene_info: ResolveInfo, **kwargs):
-        asset_keys = set(
-            AssetKey.from_graphql_input(asset_key) for asset_key in kwargs.get("assetKeys", [])
+    def resolve_assetNodes(
+        self,
+        graphene_info: ResolveInfo,
+        loadMaterializations: bool,
+        group: Optional[GrapheneAssetGroupSelector] = None,
+        pipeline: Optional[GraphenePipelineSelector] = None,
+        assetKeys: Optional[Sequence[GrapheneAssetKeyInput]] = None,
+    ) -> Sequence[GrapheneAssetNode]:
+        resolved_asset_keys = set(
+            AssetKey.from_graphql_input(cast(Mapping[str, Sequence[str]], asset_key_input))
```

Comment:
[2] this method exists for these call site, so it feels off that it requires casting like this. Seems likely worth updating the method definition.

Reply:
Good catch, it actually doesn't require a `cast`, this was left over. `GrapheneAssetKeyInput`, like all Graphene input objects, functions as an implicit `Any`.